### ISSUE TITLE
Energy Demand not working when not running all the buildings

### ISSUE THREAD

**Issue by CompFlu** [architecture-building-systems/CityEnergyAnalyst#3386]
Description
I upload a dataset with a large district of which I just want to simulate a subset. I ran the schedule-maker and radiation for the subset. However when running the energy demand for the chosen subset I get an error because the buildings I didn't simulate don't have corresponding radiation and schedule data. So it gives an error that radiation and schedule files for buildings I'm not interested in are not available (see screenshot below).

1. Go to Demand Forecasting
2. Run 'Building Solar Radiation' and 'Building Schedules' for a subset of the loaded dataset
3. Go to Demand Forecasting and run 'Building Energy Demand'

I expected to get results for the subset I wanted to run. Why is the energy demand also tried to be calculated for buildings I didn't select.

![image](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/72685806/92e5c918-8c68-49d1-941e-e931a73830f7)

 - OS and version: Windows 11
 - Version of CEA: 3.34.0

Thanks for your help!

**Comment by lilaleu** [comment]
Hello CompFlu!

Looking at your error message I suppose you skipped running schedule_maker between your radiation simulation and the demand calculation. Without a schedule, your demand can't be calculated. 

the demand module needs every other module excecuted from "Data Management" and "Demand forcasting" before it can be executed.

Best regards, let me know if it helped.

**Comment by CompFlu** [comment]
Hello lilaleu,

Thank you for your fast reply. I ran schedule_maker, however only for the datasubset I wanted to look at. So supposing I have B1000-B4000, I only ran it for the 4 buildings I wanted to look at, e.g. B1405, B1556, B3597 and B3904. So the buildings for which I wanted to run my energy demand had their respective schedule files created. 
![image](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/72685806/9bc300db-bf44-4739-901d-34b9793132e6)
I ran the building demand with the following settings:
![image](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/72685806/236d7c52-4645-41d4-a4fa-33bb59eb30f2)
And then got the error as shown in the image above. 
![image](https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/72685806/3e0c4f3d-44e5-4f07-827b-5dd9bde11ff7)

Thank you for your help and best regards.
